src/kernel/include/thd.h

```diff
@@ -20,7 +20,7 @@
 
 struct invstk_entry {
 	struct comp_info comp_info;
-	unsigned long    sp, ip; /* to return to */
+	unsigned long    sp, ip, fault_flag; /* to return to */
```

Comment:
I'm not sure how much more memory this makes each entry take.  Stack pointers should be word-aligned, so it might be better to save memory and overhead, and use the least significant bit of the `sp` to hold this info.  We can discuss this.